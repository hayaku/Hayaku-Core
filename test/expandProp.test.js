// Proxy for jest test to write abbr only once, ok
const test = (desc, func) => it(desc, () => func(desc));
test.only = (desc, func) => it.only(desc, () => func(desc));
test.skip = (desc, func) => it.skip(desc, () => func(desc));

const Hayaku = require('./../');
const getProperty = abbr => {
  const results = Hayaku.expandProp(abbr);
  return results[0] && results[0].property;
}

describe('simple one-letter prop tests:', () => {
  test('a', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('b', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('c', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('d', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('e', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('f', abbr => expect(getProperty(abbr)).toMatchSnapshot()); // `font`?
  test('g', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('h', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('i', abbr => expect(getProperty(abbr)).toMatchSnapshot()); // not used much?
  test('j', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('k', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('l', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('m', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('n', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('o', abbr => expect(getProperty(abbr)).toMatchSnapshot()); // `overflow` or not?
  test('p', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('q', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('r', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('s', abbr => expect(getProperty(abbr)).toMatchSnapshot()); // or `stroke` from svg?
  test('t', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('u', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('v', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('w', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('x', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('y', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('z', abbr => expect(getProperty(abbr)).toMatchSnapshot());
});

describe('Zen CSS compat test:', () => {
  test('pos', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  // Expands to `flex` due to recursive sorter,
  // could be fixed if `font` would be > `float`?
  // test('fl', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('cl', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('ov', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('ovx', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('ovy', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('cp', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  // test('bs', abbr => expect(getProperty(abbr)).toMatchSnapshot()); // gets border-style
  test('fxd', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('mt', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('mr', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('mb', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('ml', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('pt', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('pr', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('pb', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('pl', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('miw', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('mih', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('maw', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('mah', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('ow', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('os', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  // In those our popular > dashstarts is better as otherwise `bd` wouldn't be `border` etc.
  // Probs would help if existent is shorter?
  // test('oc', abbr => expect(getProperty(abbr)).toMatchSnapshot()); // Gets opacity?
  // test('oo', abbr => expect(getProperty(abbr)).toMatchSnapshot()); // Gets overflow?
  test('bd', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('bdcl', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('bdsp', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('bdw', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('bds', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('bdc', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('bdt', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('bdtw', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  // Probs all other bd with directions would be ok
  test('bdi', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('bg', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('bgc', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('bgi', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('br', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('bgr', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('bga', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('bgp', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('bgpy', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('bgo', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('bgz', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('bdbr', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('bsh', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('tl', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  // Some skips after this
  test('ls', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('va', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('ta', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('ti', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('ws', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('ww', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('les', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  // test('fs', abbr => expect(getProperty(abbr)).toMatchSnapshot()); // Font-style, as there is font-size hmmmm
  test('fz', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('cur', abbr => expect(getProperty(abbr)).toMatchSnapshot());
});

describe('Some other misc tests:', () => {
  test('bxs', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('po', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('fst', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('zo', abbr => expect(getProperty(abbr)).toMatchSnapshot());
  test('fl', abbr => expect(getProperty(abbr)).toMatchSnapshot());
});
