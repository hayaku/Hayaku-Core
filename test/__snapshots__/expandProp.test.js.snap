// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Some other misc tests: bxs 1`] = `"box-sizing"`;

exports[`Some other misc tests: fl 1`] = `"flex"`;

exports[`Some other misc tests: fst 1`] = `"font-style"`;

exports[`Some other misc tests: po 1`] = `"position"`;

exports[`Some other misc tests: zo 1`] = `"zoom"`;

exports[`Zen CSS compat test: bd 1`] = `"border"`;

exports[`Zen CSS compat test: bdbr 1`] = `"box-decoration-break"`;

exports[`Zen CSS compat test: bdc 1`] = `"border-color"`;

exports[`Zen CSS compat test: bdcl 1`] = `"border-collapse"`;

exports[`Zen CSS compat test: bdi 1`] = `"border-image"`;

exports[`Zen CSS compat test: bds 1`] = `"border-style"`;

exports[`Zen CSS compat test: bdsp 1`] = `"border-spacing"`;

exports[`Zen CSS compat test: bdt 1`] = `"border-top"`;

exports[`Zen CSS compat test: bdtw 1`] = `"border-top-width"`;

exports[`Zen CSS compat test: bdw 1`] = `"border-width"`;

exports[`Zen CSS compat test: bg 1`] = `"background"`;

exports[`Zen CSS compat test: bga 1`] = `"background-attachment"`;

exports[`Zen CSS compat test: bgc 1`] = `"background-color"`;

exports[`Zen CSS compat test: bgi 1`] = `"background-image"`;

exports[`Zen CSS compat test: bgo 1`] = `"background-origin"`;

exports[`Zen CSS compat test: bgp 1`] = `"background-position"`;

exports[`Zen CSS compat test: bgpy 1`] = `"background-position-y"`;

exports[`Zen CSS compat test: bgr 1`] = `"background-repeat"`;

exports[`Zen CSS compat test: bgz 1`] = `"background-size"`;

exports[`Zen CSS compat test: br 1`] = `"border-right"`;

exports[`Zen CSS compat test: bsh 1`] = `"box-shadow"`;

exports[`Zen CSS compat test: cl 1`] = `"clear"`;

exports[`Zen CSS compat test: cp 1`] = `"clip"`;

exports[`Zen CSS compat test: cur 1`] = `"cursor"`;

exports[`Zen CSS compat test: fxd 1`] = `"flex-direction"`;

exports[`Zen CSS compat test: fz 1`] = `"font-size"`;

exports[`Zen CSS compat test: les 1`] = `"letter-spacing"`;

exports[`Zen CSS compat test: ls 1`] = `"list-style"`;

exports[`Zen CSS compat test: mah 1`] = `"max-height"`;

exports[`Zen CSS compat test: maw 1`] = `"max-width"`;

exports[`Zen CSS compat test: mb 1`] = `"margin-bottom"`;

exports[`Zen CSS compat test: mih 1`] = `"min-height"`;

exports[`Zen CSS compat test: miw 1`] = `"min-width"`;

exports[`Zen CSS compat test: ml 1`] = `"margin-left"`;

exports[`Zen CSS compat test: mr 1`] = `"margin-right"`;

exports[`Zen CSS compat test: mt 1`] = `"margin-top"`;

exports[`Zen CSS compat test: os 1`] = `"outline-style"`;

exports[`Zen CSS compat test: ov 1`] = `"overflow"`;

exports[`Zen CSS compat test: ovx 1`] = `"overflow-x"`;

exports[`Zen CSS compat test: ovy 1`] = `"overflow-y"`;

exports[`Zen CSS compat test: ow 1`] = `"outline-width"`;

exports[`Zen CSS compat test: pb 1`] = `"padding-bottom"`;

exports[`Zen CSS compat test: pl 1`] = `"padding-left"`;

exports[`Zen CSS compat test: pos 1`] = `"position"`;

exports[`Zen CSS compat test: pr 1`] = `"padding-right"`;

exports[`Zen CSS compat test: pt 1`] = `"padding-top"`;

exports[`Zen CSS compat test: ta 1`] = `"text-align"`;

exports[`Zen CSS compat test: ti 1`] = `"text-indent"`;

exports[`Zen CSS compat test: tl 1`] = `"table-layout"`;

exports[`Zen CSS compat test: va 1`] = `"vertical-align"`;

exports[`Zen CSS compat test: ws 1`] = `"white-space"`;

exports[`Zen CSS compat test: ww 1`] = `"word-wrap"`;

exports[`simple one-letter prop tests: a 1`] = `"animation"`;

exports[`simple one-letter prop tests: b 1`] = `"bottom"`;

exports[`simple one-letter prop tests: c 1`] = `"color"`;

exports[`simple one-letter prop tests: d 1`] = `"display"`;

exports[`simple one-letter prop tests: e 1`] = `"empty-cells"`;

exports[`simple one-letter prop tests: f 1`] = `"float"`;

exports[`simple one-letter prop tests: g 1`] = `"grid"`;

exports[`simple one-letter prop tests: h 1`] = `"height"`;

exports[`simple one-letter prop tests: i 1`] = `"isolation"`;

exports[`simple one-letter prop tests: j 1`] = `"justify-content"`;

exports[`simple one-letter prop tests: k 1`] = `"kerning"`;

exports[`simple one-letter prop tests: l 1`] = `"left"`;

exports[`simple one-letter prop tests: m 1`] = `"margin"`;

exports[`simple one-letter prop tests: n 1`] = `undefined`;

exports[`simple one-letter prop tests: o 1`] = `"opacity"`;

exports[`simple one-letter prop tests: p 1`] = `"padding"`;

exports[`simple one-letter prop tests: q 1`] = `"quotes"`;

exports[`simple one-letter prop tests: r 1`] = `"right"`;

exports[`simple one-letter prop tests: s 1`] = `"speak"`;

exports[`simple one-letter prop tests: t 1`] = `"top"`;

exports[`simple one-letter prop tests: u 1`] = `"user-select"`;

exports[`simple one-letter prop tests: v 1`] = `"visibility"`;

exports[`simple one-letter prop tests: w 1`] = `"width"`;

exports[`simple one-letter prop tests: x 1`] = `undefined`;

exports[`simple one-letter prop tests: y 1`] = `undefined`;

exports[`simple one-letter prop tests: z 1`] = `"z-index"`;
